vyper/semantics/types/function.py

```diff
@@ -453,14 +455,19 @@ def getter_from_VariableDecl(cls, node: vy_ast.VariableDecl) -> "ContractFunctio
         ---------
         node : VariableDecl
             Vyper ast node to generate the function definition from.
+        data_loc : DataLocation
 
         Returns
         -------
         ContractFunctionT
         """
         if not node.is_public:
             raise CompilerPanic("getter generated for non-public function")
-        type_ = type_from_annotation(node.annotation, DataLocation.STORAGE)
+
+        assert data_loc not in (DataLocation.MEMORY, DataLocation.CALLDATA)
```

Comment:
why don't get get the location from the node?

Reply:
added the util `def location_from_decl`